### ISSUE TITLE
Revamp logging documentation for 0.19

### ISSUE THREAD

**Issue by noklam** [kedro-org/kedro#2664]
## Description
This is a follow-up ticket to update the documentation. Since there are many changes in logging, it makes sense to look at the whole logging documentation. Splitting this from #2642 since the PR is getting too large.

Some changes we made:
- `conf/base/logging.yml` is optional
- `conf/base/logging.yml` -> `conf/logging.yml` - it will is agnostic to environment
- Logging is setup from the beginning - project logging is optional.
- #2426 
- #2535 

This could affect our `logging.md` and some of the deployment docs and more. The best way to do it is go through the `RELEASE.md` we have made in the last couple of months and check the [closed milestone ticket](https://github.com/kedro-org/kedro/milestone/14)

# Checklist
- [ ] update the documentation
- [ ] Do a manual test to make sure everything work as expected. With/without `logging.yml`, using `KEDRO_LOGGING_CONFIG` etc.
- [ ] This need to merged after #2642 and #2520 

Also this:

> > This is a huge improvement, thank you! 🙏 I would recommend picking up https://github.com/kedro-org/kedro/issues/2520 next so then we've got all the logging code changes complete and then write the docs after that.
> > 
> > When it comes to writing docs, we should be able to make the page MUCH simpler. Basically it would be something like:
> > 1. By default kedro, uses `rich` for logging. For the full setup, look at the `default_logging.yml` file
> > 2. If you want to change this then make your own `logging.yml` file. We recommend you put that file in `conf`. Then set `KEDRO_CONFIG_LOGGING` environment variable to point to it
> > 3. Show some of the things you might like to change and the code needed to do it:
> >   a. Turn off rich tracebacks
> >   b. Turn on file logging
> >   c. Show DEBUG level messages
> > 
> > No need to explain the difference between framework-side and logging-side configuration or any of that confusing stuff any more 🎉
> > 
> > _Originally posted by @antonymilne in https://github.com/kedro-org/kedro/pull/2642#pullrequestreview-1471551105_
> >             
> 

# Follow-up
- [ ] #2671

**Comment by noklam** [comment]
Closed by #2665
